spec/compiler/parser/parser_spec.cr

```diff
@@ -22,13 +22,32 @@ private def it_parses(string, expected_node, file = __FILE__, line = __LINE__)
   end
 end
 
+private def location_to_index(string, location)
+  index = 0
+  line = 1
+  while line != location.line_number
+    index = string.index('\n', index).not_nil! + 1
+    line += 1
+  end
+  index + location.column_number - 1
+end
+
+private def source_between(string, loc, end_loc)
+  beginning = location_to_index(string, loc.not_nil!)
+  ending = location_to_index(string, end_loc.not_nil!)
+  string[beginning..ending]
+end
+
 private def assert_end_location(source, line_number = 1, column_number = source.size, file = __FILE__, line = __LINE__)
   it "gets corrects end location for #{source.inspect}", file, line do
-    parser = Parser.new("#{source}; 1")
+    string = "#{source}; 1"
+    parser = Parser.new(string)
     node = parser.parse.as(Expressions).expressions[0]
+    loc = node.location.not_nil!
     end_loc = node.end_location.not_nil!
     end_loc.line_number.should eq(line_number)
     end_loc.column_number.should eq(column_number)
+    source_between(string, loc, end_loc).should eq(source)
```

Comment:
What's the purpose of this expectation? I don't see how the source string comparison adds any relevant information to just the end location.

Reply:
It’s useful for checking if the start location is set correctly (so maybe I should move this to a separate test). However, I think the string comparison is easier to understand at a glance than line/column numbers (when the test fails), so perhaps we can delete the number comparisons altogether?